imports/client/components/HuntEditPage.tsx

```diff
@@ -0,0 +1,560 @@
+        return opt.id === value!.id;
+      })) {
+        return [noneOption, value, ...options];
+      }
+    }
+    return [noneOption, ...options];
+  }, [value, options]);
+
+  if (loading) {
+    return <div>Loading discord resources...</div>;
+  } else {
+    return (
+      <FormControl
+        id={formId}
+        as="select"
+        type="text"
+        placeholder=""
+        value={value && value.id}
+        disabled={disable}
+        onChange={onValueChanged}
+      >
+        {formOptions().map(({ id, name }) => {
+          return (
+            <option key={id} value={id}>{name}</option>
+          );
+        })}
+      </FormControl>
+    );
+  }
+};
+
+const DiscordChannelSelector = (
+  { guildId, ...rest }: DiscordSelectorParams & { guildId: string }
+) => {
+  const cacheLoading = useSubscribe('discord.cache', { type: 'channel' });
+  const loading = cacheLoading();
+
+  const { options } = useTracker(() => {
+    const discordChannels: SavedDiscordObjectType[] = DiscordCache.find({
+      type: 'channel',
+      'object.guild': guildId,
+      // We want only text channels, since those are the only ones we can bridge chat messages to.
+      'object.type': 'text',
+    }, {
+      // We want to sort them in the same order they're provided in the Discord UI.
+      sort: { 'object.rawPosition': 1 },
+      fields: { 'object.id': 1, 'object.name': 1 },
+    })
+      .map((c) => c.object as SavedDiscordObjectType);
+
+    return {
+      options: discordChannels,
+    };
+  }, [guildId]);
+  return (
+    <DiscordSelector
+      loading={loading}
+      options={options}
+      {...rest}
+    />
+  );
+};
+
+const DiscordRoleSelector = ({ guildId, ...rest }: DiscordSelectorParams & { guildId: string }) => {
+  const cacheLoading = useSubscribe('discord.cache', { type: 'role' });
+  const loading = cacheLoading();
+  const { options } = useTracker(() => {
+    const discordRoles: SavedDiscordObjectType[] = DiscordCache.find({
+      type: 'role',
+      'object.guild': guildId,
+      // The role whose id is the same as the guild is the @everyone role, don't want that
+      'object.id': { $ne: guildId },
+      // Managed roles are owned by an integration
+      'object.managed': false,
+    }, {
+      // We want to sort them in the same order they're provided in the Discord UI.
+      sort: { 'object.rawPosition': 1 },
+      fields: { 'object.id': 1, 'object.name': 1 },
+    })
+      .map((c) => c.object as SavedDiscordObjectType);
+
+    return {
+      options: discordRoles,
+    };
+  }, [guildId]);
+  return (
+    <DiscordSelector
+      loading={loading}
+      options={options}
+      {...rest}
+    />
+  );
+};
+
+const HuntEditPage = () => {
+  const huntId = useParams<{ huntId: string }>().huntId;
+  const hunt = useTracker(() => (huntId ? Hunts.findOne(huntId) : null), [huntId]);
+
+  useBreadcrumb({ title: huntId ? 'Edit Hunt' : 'Create Hunt', path: `/hunts/${huntId ? `${huntId}/edit` : 'new'}` });
```

Comment:
As written, this results in there being no clickable breadcrumb for `/hunts/new`, which feels a little odd to me.

For the case where we're creating a new hunt, I think I'd lightly prefer if we had two breadcrumbs here -- one "Hunts", linking to `/hunts`, and then maybe "Create" (at `/hunts/new`).  Thoughts?

Reply:
I'll poke at this in a bit. The reason I didn't add a second breadcrumb initially is that you can't call hooks conditionally. I can try to modify `useBreadcrumb` to take `object | null` and not render the breadcrumb if null is passed in (similar to how `useSubscribe` works), but I'm not sure if that will actually work with `useBreadcrumb`'s weird state tracking.

Reply:
OK managed to get this working by just rearranging the route map a bit. I'll push in a sec